guaranteed-message-delivery/README.md

```diff
@@ -0,0 +1,390 @@
+
+    ```ballerina
+    type MessageContent record {|
+        *ReservationRequest;
+        string category;
+    |};
+
+    type RabbitMqMessage record {|
+        *rabbitmq:AnydataMessage;
+        MessageContent content;
+    |};
+    ```
+
+    > **Note:**
+    > Using [record type inclusion](https://ballerina.io/learn/by-example/type-inclusion-for-records/) allows including all the fields from the included record along with the defined fields.
+
+9. Define configurations for the SMS service endpoints (e.g. `fromNumber`, `accountSId`, `authToken`) as [configurable variables](https://ballerina.io/learn/by-example/#configurability).
+
+    ```ballerina
+    configurable string fromNumber = ?;
+    configurable string accountSId = ?;
+    configurable string authToken = ?;
+    ```
+
+10. Implement the consumer service.
+
+    ```ballerina
+    @rabbitmq:ServiceConfig {
+        queueName
+    }
+    service rabbitmq:Service on new rabbitmq:Listener(rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT) {
+        remote function onMessage(RabbitMqMessage message) {
+            MessageContent content = message.content;
+            string hospital = content.hospital;
+            string patientName = content.patient.name;
+            string doctor = content.doctor;
+
+            ReservationResponse|http:ClientError reservationResponse = 
+                hospitalBackend->/[content.hospital_id]/categories/[content.category]/reserve.post({
+                    patient: content.patient,
+                    doctor,
+                    hospital,
+                    appointment_date: content.appointment_date
+                });
+
+            string smsBody;
+            if reservationResponse is http:ClientError {
+                log:printError("Reservation request failed", patient = patientName,
+                                                            doctor = doctor,
+                                                            hospital = hospital);
+                smsBody = string `Dear ${patientName
+                            }, your appointment request at ${hospital
+                            } failed. Please try again.`;
+            } else {
+                smsBody = string `Dear ${patientName
+                            }, your appointment has been accepted at ${hospital
+                            }. Appointment No: ${reservationResponse.appointmentNumber}`;
+            }
+
+            twilio:SmsResponse|error smsApiStatus = twilioEp->sendSms(fromNumber, content.patient.phone, smsBody);
+
+            if smsApiStatus !is twilio:SmsResponse {
+                log:printError("Failed to send an SMS message", smsApiStatus, phoneNo = content.patient.phone);
+            }
+        }
+    }
+    ```
+
+    -  Create the `rabbitmq:Service` listening on `rabbitmq:Listener` and remote function named `onMessage` which takes a message as an argument.
```

Comment:
```suggestion
    -  Create the `rabbitmq:Service` service listening on a `rabbitmq:Listener` listener with the  `onMessage` remote method that gets called ....
```

We need to describe vs just stating where possible, so that readers understand the concepts.